src/buffer_ops/copy.c

```diff
@@ -259,6 +259,12 @@ bool pmix_value_cmp(pmix_value_t *p, pmix_value_t *p1)
         case PMIX_STRING:
             rc = strcmp(p->data.string, p1->data.string);
             break;
+        case PMIX_COMPRESSED_STRING:
+            if (p->data.bo.size != p1->data.bo.size) {
```

Comment:
My concern is that we will most probably forget about this conversation when/if we will decide to use it.
Doing byte comparison may save us some debugging cycles.